{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wed Jul 19 13:34:30 PDT 2017\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Software:\n",
      "\n",
      "    System Software Overview:\n",
      "\n",
      "      System Version: OS X 10.9.5 (13F34)\n",
      "      Kernel Version: Darwin 13.4.0\n",
      "      Boot Volume: Hummingbird\n",
      "      Boot Mode: Normal\n",
      "      Computer Name: hummingbird\n",
      "      User Name: srlab (srlab)\n",
      "      Secure Virtual Memory: Enabled\n",
      "      Time since boot: 43 minutes\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "system_profiler SPSoftwareDataType"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is all happening in scaphapoda on Eagle"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Blast practice *C. gigas* with Swiss Prot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Source: https://github.com/sr320/nb-2017/blob/master/C_gigas/data/Cg_Gigaton_proteins.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Finding location of BLAST database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 8\n",
      "drwxrwxr-x  21 root   admin   782 Nov  9  2016 Data\n",
      "lrwxr-xr-x   1 root   admin     1 Jul 19 12:54 Hummingbird -> /\n",
      "drwxr-xr-x@ 30 root   wheel  1088 Jan  8  2014 Mirror\n",
      "drwxrwxr-x  15 srlab  staff   578 Sep 16  2016 toaster\n",
      "drwx------@  1 srlab  staff   500 Oct  9  2015 web\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -l /Volumes/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 0\n",
      "drwxr-xr-x     9 srlab   admin    306 Jan 28  2014 CLC_local_bird\n",
      "drwxr-xr-x     3 srlab   admin    102 Oct 17  2016 Hollie\n",
      "drwxr-xr-x     6 root    admin    204 Jan 28  2015 Lisa\n",
      "drwxr-xr-x     7 Sam     admin    238 Apr  7  2015 Sam\n",
      "drwxr-xr-x     9 srlab   admin    306 Nov  9  2016 Sean\n",
      "drwxr-xr-x@   48 srlab   admin   1632 Nov 22  2016 Ubuntu 64-bit 16.04.1.vmwarevm\n",
      "drwxrwxrwx  1255 srlab   admin  42670 Sep 16  2016 blast_db\n",
      "drwxr-xr-x    86 root    admin   2924 Jul  3  2014 galaxy-dist\n",
      "drwxr-xr-x    10 gilesg  admin    340 Aug 11  2014 giles\n",
      "drwxrwxrwx     3 srlab   admin    102 Sep 14  2016 sr320_scratch\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -l /Volumes/Data/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Crassostrea_gigas.GCA_000297895.1.24.fa.nhr\n",
      "Crassostrea_gigas.GCA_000297895.1.24.fa.nin\n",
      "Crassostrea_gigas.GCA_000297895.1.24.fa.nog\n",
      "Crassostrea_gigas.GCA_000297895.1.24.fa.nsd\n",
      "Crassostrea_gigas.GCA_000297895.1.24.fa.nsi\n",
      "Crassostrea_gigas.GCA_000297895.1.24.fa.nsq\n",
      "Strongylocentrotus_purpuratus.GCA_000002235.2.22.pep.all.fa\n",
      "Strongylocentrotus_purpuratus.GCA_000002235.2.22.pep.all.phr\n",
      "Strongylocentrotus_purpuratus.GCA_000002235.2.22.pep.all.pin\n",
      "Strongylocentrotus_purpuratus.GCA_000002235.2.22.pep.all.psq\n",
      "cdd_delta.aux\n",
      "cdd_delta.loo\n",
      "cdd_delta.obsr\n",
      "cdd_delta.phr\n",
      "cdd_delta.pin\n",
      "cdd_delta.psd\n",
      "cdd_delta.psi\n",
      "cdd_delta.psq\n",
      "cdd_delta.rps\n",
      "cdd_delta.wcounts\n",
      "env_nr.00.phd\n",
      "env_nr.00.phi\n",
      "env_nr.00.phr\n",
      "env_nr.00.pin\n",
      "env_nr.00.pnd\n",
      "env_nr.00.pni\n",
      "env_nr.00.pog\n",
      "env_nr.00.ppd\n",
      "env_nr.00.ppi\n",
      "env_nr.00.psd\n",
      "env_nr.00.psi\n",
      "env_nr.00.psq\n",
      "env_nr.01.phd\n",
      "env_nr.01.phi\n",
      "env_nr.01.phr\n",
      "env_nr.01.pin\n",
      "env_nr.01.pnd\n",
      "env_nr.01.pni\n",
      "env_nr.01.pog\n",
      "env_nr.01.ppd\n",
      "env_nr.01.ppi\n",
      "env_nr.01.psd\n",
      "env_nr.01.psi\n",
      "env_nr.01.psq\n",
      "env_nr.pal\n",
      "env_nt.00.nhd\n",
      "env_nt.00.nhi\n",
      "env_nt.00.nhr\n",
      "env_nt.00.nin\n",
      "env_nt.00.nnd\n",
      "env_nt.00.nni\n",
      "env_nt.00.nog\n",
      "env_nt.00.nsd\n",
      "env_nt.00.nsi\n",
      "env_nt.00.nsq\n",
      "env_nt.01.nhd\n",
      "env_nt.01.nhi\n",
      "env_nt.01.nhr\n",
      "env_nt.01.nin\n",
      "env_nt.01.nnd\n",
      "env_nt.01.nni\n",
      "env_nt.01.nsd\n",
      "env_nt.01.nsi\n",
      "env_nt.01.nsq\n",
      "env_nt.02.nhd\n",
      "env_nt.02.nhi\n",
      "env_nt.02.nhr\n",
      "env_nt.02.nin\n",
      "env_nt.02.nnd\n",
      "env_nt.02.nni\n",
      "env_nt.02.nog\n",
      "env_nt.02.nsd\n",
      "env_nt.02.nsi\n",
      "env_nt.02.nsq\n",
      "env_nt.03.nhd\n",
      "env_nt.03.nhi\n",
      "env_nt.03.nhr\n",
      "env_nt.03.nin\n",
      "env_nt.03.nnd\n",
      "env_nt.03.nni\n",
      "env_nt.03.nog\n",
      "env_nt.03.nsd\n",
      "env_nt.03.nsi\n",
      "env_nt.03.nsq\n",
      "env_nt.nal\n",
      "est_others.00.nhr\n",
      "est_others.00.nin\n",
      "est_others.00.nnd\n",
      "est_others.00.nni\n",
      "est_others.00.nog\n",
      "est_others.00.nsd\n",
      "est_others.00.nsi\n",
      "est_others.00.nsq\n",
      "est_others.01.nhr\n",
      "est_others.01.nin\n",
      "est_others.01.nnd\n",
      "est_others.01.nni\n",
      "est_others.01.nsd\n",
      "est_others.01.nsi\n",
      "est_others.01.nsq\n",
      "est_others.02.nhr\n",
      "est_others.02.nin\n",
      "est_others.02.nnd\n",
      "est_others.02.nni\n",
      "est_others.02.nog\n",
      "est_others.02.nsd\n",
      "est_others.02.nsi\n",
      "est_others.02.nsq\n",
      "est_others.03.nhr\n",
      "est_others.03.nin\n",
      "est_others.03.nnd\n",
      "est_others.03.nni\n",
      "est_others.03.nog\n",
      "est_others.03.nsd\n",
      "est_others.03.nsi\n",
      "est_others.03.nsq\n",
      "est_others.04.nhr\n",
      "est_others.04.nin\n",
      "est_others.04.nnd\n",
      "est_others.04.nni\n",
      "est_others.04.nog\n",
      "est_others.04.nsd\n",
      "est_others.04.nsi\n",
      "est_others.04.nsq\n",
      "est_others.05.nhr\n",
      "est_others.05.nin\n",
      "est_others.05.nnd\n",
      "est_others.05.nni\n",
      "est_others.05.nsd\n",
      "est_others.05.nsi\n",
      "est_others.05.nsq\n",
      "est_others.06.nhr\n",
      "est_others.06.nin\n",
      "est_others.06.nnd\n",
      "est_others.06.nni\n",
      "est_others.06.nog\n",
      "est_others.06.nsd\n",
      "est_others.06.nsi\n",
      "est_others.06.nsq\n",
      "est_others.07.nhr\n",
      "est_others.07.nin\n",
      "est_others.07.nnd\n",
      "est_others.07.nni\n",
      "est_others.07.nsd\n",
      "est_others.07.nsi\n",
      "est_others.07.nsq\n",
      "est_others.08.nhr\n",
      "est_others.08.nin\n",
      "est_others.08.nnd\n",
      "est_others.08.nni\n",
      "est_others.08.nog\n",
      "est_others.08.nsd\n",
      "est_others.08.nsi\n",
      "est_others.08.nsq\n",
      "est_others.09.nhr\n",
      "est_others.09.nin\n",
      "est_others.09.nnd\n",
      "est_others.09.nni\n",
      "est_others.09.nsd\n",
      "est_others.09.nsi\n",
      "est_others.09.nsq\n",
      "est_others.10.nhr\n",
      "est_others.10.nin\n",
      "est_others.10.nnd\n",
      "est_others.10.nni\n",
      "est_others.10.nsd\n",
      "est_others.10.nsi\n",
      "est_others.10.nsq\n",
      "est_others.nal\n",
      "nr.00.phd\n",
      "nr.00.phi\n",
      "nr.00.phr\n",
      "nr.00.pin\n",
      "nr.00.pnd\n",
      "nr.00.pni\n",
      "nr.00.pog\n",
      "nr.00.ppd\n",
      "nr.00.ppi\n",
      "nr.00.psd\n",
      "nr.00.psi\n",
      "nr.00.psq\n",
      "nr.01.phd\n",
      "nr.01.phi\n",
      "nr.01.phr\n",
      "nr.01.pin\n",
      "nr.01.pnd\n",
      "nr.01.pni\n",
      "nr.01.pog\n",
      "nr.01.ppd\n",
      "nr.01.ppi\n",
      "nr.01.psd\n",
      "nr.01.psi\n",
      "nr.01.psq\n",
      "nr.02.phd\n",
      "nr.02.phi\n",
      "nr.02.phr\n",
      "nr.02.pin\n",
      "nr.02.pnd\n",
      "nr.02.pni\n",
      "nr.02.pog\n",
      "nr.02.ppd\n",
      "nr.02.ppi\n",
      "nr.02.psd\n",
      "nr.02.psi\n",
      "nr.02.psq\n",
      "nr.03.phd\n",
      "nr.03.phi\n",
      "nr.03.phr\n",
      "nr.03.pin\n",
      "nr.03.pnd\n",
      "nr.03.pni\n",
      "nr.03.pog\n",
      "nr.03.ppd\n",
      "nr.03.ppi\n",
      "nr.03.psd\n",
      "nr.03.psi\n",
      "nr.03.psq\n",
      "nr.04.phd\n",
      "nr.04.phi\n",
      "nr.04.phr\n",
      "nr.04.pin\n",
      "nr.04.pnd\n",
      "nr.04.pni\n",
      "nr.04.pog\n",
      "nr.04.ppd\n",
      "nr.04.ppi\n",
      "nr.04.psd\n",
      "nr.04.psi\n",
      "nr.04.psq\n",
      "nr.05.phd\n",
      "nr.05.phi\n",
      "nr.05.phr\n",
      "nr.05.pin\n",
      "nr.05.pnd\n",
      "nr.05.pni\n",
      "nr.05.pog\n",
      "nr.05.ppd\n",
      "nr.05.ppi\n",
      "nr.05.psd\n",
      "nr.05.psi\n",
      "nr.05.psq\n",
      "nr.06.phd\n",
      "nr.06.phi\n",
      "nr.06.phr\n",
      "nr.06.pin\n",
      "nr.06.pnd\n",
      "nr.06.pni\n",
      "nr.06.pog\n",
      "nr.06.ppd\n",
      "nr.06.ppi\n",
      "nr.06.psd\n",
      "nr.06.psi\n",
      "nr.06.psq\n",
      "nr.07.phd\n",
      "nr.07.phi\n",
      "nr.07.phr\n",
      "nr.07.pin\n",
      "nr.07.pnd\n",
      "nr.07.pni\n",
      "nr.07.pog\n",
      "nr.07.ppd\n",
      "nr.07.ppi\n",
      "nr.07.psd\n",
      "nr.07.psi\n",
      "nr.07.psq\n",
      "nr.08.phd\n",
      "nr.08.phi\n",
      "nr.08.phr\n",
      "nr.08.pin\n",
      "nr.08.pnd\n",
      "nr.08.pni\n",
      "nr.08.pog\n",
      "nr.08.ppd\n",
      "nr.08.ppi\n",
      "nr.08.psd\n",
      "nr.08.psi\n",
      "nr.08.psq\n",
      "nr.09.phd\n",
      "nr.09.phi\n",
      "nr.09.phr\n",
      "nr.09.pin\n",
      "nr.09.pnd\n",
      "nr.09.pni\n",
      "nr.09.pog\n",
      "nr.09.ppd\n",
      "nr.09.ppi\n",
      "nr.09.psd\n",
      "nr.09.psi\n",
      "nr.09.psq\n",
      "nr.10.phd\n",
      "nr.10.phi\n",
      "nr.10.phr\n",
      "nr.10.pin\n",
      "nr.10.pnd\n",
      "nr.10.pni\n",
      "nr.10.pog\n",
      "nr.10.ppd\n",
      "nr.10.ppi\n",
      "nr.10.psd\n",
      "nr.10.psi\n",
      "nr.10.psq\n",
      "nr.11.phd\n",
      "nr.11.phi\n",
      "nr.11.phr\n",
      "nr.11.pin\n",
      "nr.11.pnd\n",
      "nr.11.pni\n",
      "nr.11.pog\n",
      "nr.11.ppd\n",
      "nr.11.ppi\n",
      "nr.11.psd\n",
      "nr.11.psi\n",
      "nr.11.psq\n",
      "nr.12.phd\n",
      "nr.12.phi\n",
      "nr.12.phr\n",
      "nr.12.pin\n",
      "nr.12.pnd\n",
      "nr.12.pni\n",
      "nr.12.pog\n",
      "nr.12.ppd\n",
      "nr.12.ppi\n",
      "nr.12.psd\n",
      "nr.12.psi\n",
      "nr.12.psq\n",
      "nr.13.phd\n",
      "nr.13.phi\n",
      "nr.13.phr\n",
      "nr.13.pin\n",
      "nr.13.pnd\n",
      "nr.13.pni\n",
      "nr.13.pog\n",
      "nr.13.ppd\n",
      "nr.13.ppi\n",
      "nr.13.psd\n",
      "nr.13.psi\n",
      "nr.13.psq\n",
      "nr.pal\n",
      "nt.00.nhd\n",
      "nt.00.nhi\n",
      "nt.00.nhr\n",
      "nt.00.nin\n",
      "nt.00.nnd\n",
      "nt.00.nni\n",
      "nt.00.nog\n",
      "nt.00.nsd\n",
      "nt.00.nsi\n",
      "nt.00.nsq\n",
      "nt.01.nhd\n",
      "nt.01.nhi\n",
      "nt.01.nhr\n",
      "nt.01.nin\n",
      "nt.01.nnd\n",
      "nt.01.nni\n",
      "nt.01.nog\n",
      "nt.01.nsd\n",
      "nt.01.nsi\n",
      "nt.01.nsq\n",
      "nt.02.nhd\n",
      "nt.02.nhi\n",
      "nt.02.nhr\n",
      "nt.02.nin\n",
      "nt.02.nnd\n",
      "nt.02.nni\n",
      "nt.02.nog\n",
      "nt.02.nsd\n",
      "nt.02.nsi\n",
      "nt.02.nsq\n",
      "nt.03.nhd\n",
      "nt.03.nhi\n",
      "nt.03.nhr\n",
      "nt.03.nin\n",
      "nt.03.nnd\n",
      "nt.03.nni\n",
      "nt.03.nog\n",
      "nt.03.nsd\n",
      "nt.03.nsi\n",
      "nt.03.nsq\n",
      "nt.04.nhd\n",
      "nt.04.nhi\n",
      "nt.04.nhr\n",
      "nt.04.nin\n",
      "nt.04.nnd\n",
      "nt.04.nni\n",
      "nt.04.nog\n",
      "nt.04.nsd\n",
      "nt.04.nsi\n",
      "nt.04.nsq\n",
      "nt.05.nhd\n",
      "nt.05.nhi\n",
      "nt.05.nhr\n",
      "nt.05.nin\n",
      "nt.05.nnd\n",
      "nt.05.nni\n",
      "nt.05.nog\n",
      "nt.05.nsd\n",
      "nt.05.nsi\n",
      "nt.05.nsq\n",
      "nt.06.nhd\n",
      "nt.06.nhi\n",
      "nt.06.nhr\n",
      "nt.06.nin\n",
      "nt.06.nnd\n",
      "nt.06.nni\n",
      "nt.06.nog\n",
      "nt.06.nsd\n",
      "nt.06.nsi\n",
      "nt.06.nsq\n",
      "nt.07.nhd\n",
      "nt.07.nhi\n",
      "nt.07.nhr\n",
      "nt.07.nin\n",
      "nt.07.nnd\n",
      "nt.07.nni\n",
      "nt.07.nog\n",
      "nt.07.nsd\n",
      "nt.07.nsi\n",
      "nt.07.nsq\n",
      "nt.08.nhd\n",
      "nt.08.nhi\n",
      "nt.08.nhr\n",
      "nt.08.nin\n",
      "nt.08.nnd\n",
      "nt.08.nni\n",
      "nt.08.nog\n",
      "nt.08.nsd\n",
      "nt.08.nsi\n",
      "nt.08.nsq\n",
      "nt.09.nhd\n",
      "nt.09.nhi\n",
      "nt.09.nhr\n",
      "nt.09.nin\n",
      "nt.09.nnd\n",
      "nt.09.nni\n",
      "nt.09.nog\n",
      "nt.09.nsd\n",
      "nt.09.nsi\n",
      "nt.09.nsq\n",
      "nt.10.nhd\n",
      "nt.10.nhi\n",
      "nt.10.nhr\n",
      "nt.10.nin\n",
      "nt.10.nnd\n",
      "nt.10.nni\n",
      "nt.10.nog\n",
      "nt.10.nsd\n",
      "nt.10.nsi\n",
      "nt.10.nsq\n",
      "nt.11.nhd\n",
      "nt.11.nhi\n",
      "nt.11.nhr\n",
      "nt.11.nin\n",
      "nt.11.nnd\n",
      "nt.11.nni\n",
      "nt.11.nog\n",
      "nt.11.nsd\n",
      "nt.11.nsi\n",
      "nt.11.nsq\n",
      "nt.12.nhd\n",
      "nt.12.nhi\n",
      "nt.12.nhr\n",
      "nt.12.nin\n",
      "nt.12.nnd\n",
      "nt.12.nni\n",
      "nt.12.nog\n",
      "nt.12.nsd\n",
      "nt.12.nsi\n",
      "nt.12.nsq\n",
      "nt.13.nhd\n",
      "nt.13.nhi\n",
      "nt.13.nhr\n",
      "nt.13.nin\n",
      "nt.13.nnd\n",
      "nt.13.nni\n",
      "nt.13.nog\n",
      "nt.13.nsd\n",
      "nt.13.nsi\n",
      "nt.13.nsq\n",
      "nt.14.nhd\n",
      "nt.14.nhi\n",
      "nt.14.nhr\n",
      "nt.14.nin\n",
      "nt.14.nnd\n",
      "nt.14.nni\n",
      "nt.14.nog\n",
      "nt.14.nsd\n",
      "nt.14.nsi\n",
      "nt.14.nsq\n",
      "nt.15.nhd\n",
      "nt.15.nhi\n",
      "nt.15.nhr\n",
      "nt.15.nin\n",
      "nt.15.nnd\n",
      "nt.15.nni\n",
      "nt.15.nog\n",
      "nt.15.nsd\n",
      "nt.15.nsi\n",
      "nt.15.nsq\n",
      "nt.nal\n",
      "pdbaa.00.msk\n",
      "pdbaa.00.pal\n",
      "pdbaa.01.msk\n",
      "pdbaa.01.pal\n",
      "pdbaa.02.msk\n",
      "pdbaa.02.pal\n",
      "pdbaa.03.msk\n",
      "pdbaa.03.pal\n",
      "pdbaa.04.msk\n",
      "pdbaa.04.pal\n",
      "pdbaa.05.msk\n",
      "pdbaa.05.pal\n",
      "pdbaa.06.msk\n",
      "pdbaa.06.pal\n",
      "pdbaa.07.msk\n",
      "pdbaa.07.pal\n",
      "pdbaa.08.msk\n",
      "pdbaa.08.pal\n",
      "pdbaa.09.msk\n",
      "pdbaa.09.pal\n",
      "pdbaa.10.msk\n",
      "pdbaa.10.pal\n",
      "pdbaa.11.msk\n",
      "pdbaa.11.pal\n",
      "pdbaa.12.msk\n",
      "pdbaa.12.pal\n",
      "pdbaa.13.pal\n",
      "pdbaa.pal\n",
      "refseq_genomic.00.nhr\n",
      "refseq_genomic.00.nin\n",
      "refseq_genomic.00.nnd\n",
      "refseq_genomic.00.nni\n",
      "refseq_genomic.00.nog\n",
      "refseq_genomic.00.nsd\n",
      "refseq_genomic.00.nsi\n",
      "refseq_genomic.00.nsq\n",
      "refseq_genomic.01.nhr\n",
      "refseq_genomic.01.nin\n",
      "refseq_genomic.01.nnd\n",
      "refseq_genomic.01.nni\n",
      "refseq_genomic.01.nog\n",
      "refseq_genomic.01.nsd\n",
      "refseq_genomic.01.nsi\n",
      "refseq_genomic.01.nsq\n",
      "refseq_genomic.02.nhr\n",
      "refseq_genomic.02.nin\n",
      "refseq_genomic.02.nnd\n",
      "refseq_genomic.02.nni\n",
      "refseq_genomic.02.nog\n",
      "refseq_genomic.02.nsd\n",
      "refseq_genomic.02.nsi\n",
      "refseq_genomic.02.nsq\n",
      "refseq_genomic.03.nhr\n",
      "refseq_genomic.03.nin\n",
      "refseq_genomic.03.nnd\n",
      "refseq_genomic.03.nni\n",
      "refseq_genomic.03.nog\n",
      "refseq_genomic.03.nsd\n",
      "refseq_genomic.03.nsi\n",
      "refseq_genomic.03.nsq\n",
      "refseq_genomic.04.nhr\n",
      "refseq_genomic.04.nin\n",
      "refseq_genomic.04.nnd\n",
      "refseq_genomic.04.nni\n",
      "refseq_genomic.04.nog\n",
      "refseq_genomic.04.nsd\n",
      "refseq_genomic.04.nsi\n",
      "refseq_genomic.04.nsq\n",
      "refseq_genomic.05.nhr\n",
      "refseq_genomic.05.nin\n",
      "refseq_genomic.05.nnd\n",
      "refseq_genomic.05.nni\n",
      "refseq_genomic.05.nog\n",
      "refseq_genomic.05.nsd\n",
      "refseq_genomic.05.nsi\n",
      "refseq_genomic.05.nsq\n",
      "refseq_genomic.06.nhr\n",
      "refseq_genomic.06.nin\n",
      "refseq_genomic.06.nnd\n",
      "refseq_genomic.06.nni\n",
      "refseq_genomic.06.nog\n",
      "refseq_genomic.06.nsd\n",
      "refseq_genomic.06.nsi\n",
      "refseq_genomic.06.nsq\n",
      "refseq_genomic.07.nhr\n",
      "refseq_genomic.07.nin\n",
      "refseq_genomic.07.nnd\n",
      "refseq_genomic.07.nni\n",
      "refseq_genomic.07.nog\n",
      "refseq_genomic.07.nsd\n",
      "refseq_genomic.07.nsi\n",
      "refseq_genomic.07.nsq\n",
      "refseq_genomic.08.nhr\n",
      "refseq_genomic.08.nin\n",
      "refseq_genomic.08.nnd\n",
      "refseq_genomic.08.nni\n",
      "refseq_genomic.08.nog\n",
      "refseq_genomic.08.nsd\n",
      "refseq_genomic.08.nsi\n",
      "refseq_genomic.08.nsq\n",
      "refseq_genomic.09.nhr\n",
      "refseq_genomic.09.nin\n",
      "refseq_genomic.09.nnd\n",
      "refseq_genomic.09.nni\n",
      "refseq_genomic.09.nog\n",
      "refseq_genomic.09.nsd\n",
      "refseq_genomic.09.nsi\n",
      "refseq_genomic.09.nsq\n",
      "refseq_genomic.10.nhr\n",
      "refseq_genomic.10.nin\n",
      "refseq_genomic.10.nnd\n",
      "refseq_genomic.10.nni\n",
      "refseq_genomic.10.nog\n",
      "refseq_genomic.10.nsd\n",
      "refseq_genomic.10.nsi\n",
      "refseq_genomic.10.nsq\n",
      "refseq_genomic.11.nhr\n",
      "refseq_genomic.11.nin\n",
      "refseq_genomic.11.nnd\n",
      "refseq_genomic.11.nni\n",
      "refseq_genomic.11.nog\n",
      "refseq_genomic.11.nsd\n",
      "refseq_genomic.11.nsi\n",
      "refseq_genomic.11.nsq\n",
      "refseq_genomic.12.nhr\n",
      "refseq_genomic.12.nin\n",
      "refseq_genomic.12.nnd\n",
      "refseq_genomic.12.nni\n",
      "refseq_genomic.12.nog\n",
      "refseq_genomic.12.nsd\n",
      "refseq_genomic.12.nsi\n",
      "refseq_genomic.12.nsq\n",
      "refseq_genomic.13.nhr\n",
      "refseq_genomic.13.nin\n",
      "refseq_genomic.13.nnd\n",
      "refseq_genomic.13.nni\n",
      "refseq_genomic.13.nog\n",
      "refseq_genomic.13.nsd\n",
      "refseq_genomic.13.nsi\n",
      "refseq_genomic.13.nsq\n",
      "refseq_genomic.14.nhr\n",
      "refseq_genomic.14.nin\n",
      "refseq_genomic.14.nnd\n",
      "refseq_genomic.14.nni\n",
      "refseq_genomic.14.nog\n",
      "refseq_genomic.14.nsd\n",
      "refseq_genomic.14.nsi\n",
      "refseq_genomic.14.nsq\n",
      "refseq_genomic.15.nhr\n",
      "refseq_genomic.15.nin\n",
      "refseq_genomic.15.nnd\n",
      "refseq_genomic.15.nni\n",
      "refseq_genomic.15.nog\n",
      "refseq_genomic.15.nsd\n",
      "refseq_genomic.15.nsi\n",
      "refseq_genomic.15.nsq\n",
      "refseq_genomic.16.nhr\n",
      "refseq_genomic.16.nin\n",
      "refseq_genomic.16.nnd\n",
      "refseq_genomic.16.nni\n",
      "refseq_genomic.16.nog\n",
      "refseq_genomic.16.nsd\n",
      "refseq_genomic.16.nsi\n",
      "refseq_genomic.16.nsq\n",
      "refseq_genomic.17.nhr\n",
      "refseq_genomic.17.nin\n",
      "refseq_genomic.17.nnd\n",
      "refseq_genomic.17.nni\n",
      "refseq_genomic.17.nog\n",
      "refseq_genomic.17.nsd\n",
      "refseq_genomic.17.nsi\n",
      "refseq_genomic.17.nsq\n",
      "refseq_genomic.18.nhr\n",
      "refseq_genomic.18.nin\n",
      "refseq_genomic.18.nnd\n",
      "refseq_genomic.18.nni\n",
      "refseq_genomic.18.nog\n",
      "refseq_genomic.18.nsd\n",
      "refseq_genomic.18.nsi\n",
      "refseq_genomic.18.nsq\n",
      "refseq_genomic.19.nhr\n",
      "refseq_genomic.19.nin\n",
      "refseq_genomic.19.nnd\n",
      "refseq_genomic.19.nni\n",
      "refseq_genomic.19.nog\n",
      "refseq_genomic.19.nsd\n",
      "refseq_genomic.19.nsi\n",
      "refseq_genomic.19.nsq\n",
      "refseq_genomic.20.nhr\n",
      "refseq_genomic.20.nin\n",
      "refseq_genomic.20.nnd\n",
      "refseq_genomic.20.nni\n",
      "refseq_genomic.20.nog\n",
      "refseq_genomic.20.nsd\n",
      "refseq_genomic.20.nsi\n",
      "refseq_genomic.20.nsq\n",
      "refseq_genomic.21.nhr\n",
      "refseq_genomic.21.nin\n",
      "refseq_genomic.21.nnd\n",
      "refseq_genomic.21.nni\n",
      "refseq_genomic.21.nog\n",
      "refseq_genomic.21.nsd\n",
      "refseq_genomic.21.nsi\n",
      "refseq_genomic.21.nsq\n",
      "refseq_genomic.22.nhr\n",
      "refseq_genomic.22.nin\n",
      "refseq_genomic.22.nnd\n",
      "refseq_genomic.22.nni\n",
      "refseq_genomic.22.nog\n",
      "refseq_genomic.22.nsd\n",
      "refseq_genomic.22.nsi\n",
      "refseq_genomic.22.nsq\n",
      "refseq_genomic.23.nhr\n",
      "refseq_genomic.23.nin\n",
      "refseq_genomic.23.nnd\n",
      "refseq_genomic.23.nni\n",
      "refseq_genomic.23.nog\n",
      "refseq_genomic.23.nsd\n",
      "refseq_genomic.23.nsi\n",
      "refseq_genomic.23.nsq\n",
      "refseq_genomic.24.nhr\n",
      "refseq_genomic.24.nin\n",
      "refseq_genomic.24.nnd\n",
      "refseq_genomic.24.nni\n",
      "refseq_genomic.24.nog\n",
      "refseq_genomic.24.nsd\n",
      "refseq_genomic.24.nsi\n",
      "refseq_genomic.24.nsq\n",
      "refseq_genomic.25.nhr\n",
      "refseq_genomic.25.nin\n",
      "refseq_genomic.25.nnd\n",
      "refseq_genomic.25.nni\n",
      "refseq_genomic.25.nog\n",
      "refseq_genomic.25.nsd\n",
      "refseq_genomic.25.nsi\n",
      "refseq_genomic.25.nsq\n",
      "refseq_genomic.26.nhr\n",
      "refseq_genomic.26.nin\n",
      "refseq_genomic.26.nnd\n",
      "refseq_genomic.26.nni\n",
      "refseq_genomic.26.nog\n",
      "refseq_genomic.26.nsd\n",
      "refseq_genomic.26.nsi\n",
      "refseq_genomic.26.nsq\n",
      "refseq_genomic.27.nhr\n",
      "refseq_genomic.27.nin\n",
      "refseq_genomic.27.nnd\n",
      "refseq_genomic.27.nni\n",
      "refseq_genomic.27.nog\n",
      "refseq_genomic.27.nsd\n",
      "refseq_genomic.27.nsi\n",
      "refseq_genomic.27.nsq\n",
      "refseq_genomic.28.nhr\n",
      "refseq_genomic.28.nin\n",
      "refseq_genomic.28.nnd\n",
      "refseq_genomic.28.nni\n",
      "refseq_genomic.28.nog\n",
      "refseq_genomic.28.nsd\n",
      "refseq_genomic.28.nsi\n",
      "refseq_genomic.28.nsq\n",
      "refseq_genomic.29.nhr\n",
      "refseq_genomic.29.nin\n",
      "refseq_genomic.29.nnd\n",
      "refseq_genomic.29.nni\n",
      "refseq_genomic.29.nog\n",
      "refseq_genomic.29.nsd\n",
      "refseq_genomic.29.nsi\n",
      "refseq_genomic.29.nsq\n",
      "refseq_genomic.30.nhr\n",
      "refseq_genomic.30.nin\n",
      "refseq_genomic.30.nnd\n",
      "refseq_genomic.30.nni\n",
      "refseq_genomic.30.nog\n",
      "refseq_genomic.30.nsd\n",
      "refseq_genomic.30.nsi\n",
      "refseq_genomic.30.nsq\n",
      "refseq_genomic.31.nhr\n",
      "refseq_genomic.31.nin\n",
      "refseq_genomic.31.nnd\n",
      "refseq_genomic.31.nni\n",
      "refseq_genomic.31.nog\n",
      "refseq_genomic.31.nsd\n",
      "refseq_genomic.31.nsi\n",
      "refseq_genomic.31.nsq\n",
      "refseq_genomic.32.nhr\n",
      "refseq_genomic.32.nin\n",
      "refseq_genomic.32.nnd\n",
      "refseq_genomic.32.nni\n",
      "refseq_genomic.32.nog\n",
      "refseq_genomic.32.nsd\n",
      "refseq_genomic.32.nsi\n",
      "refseq_genomic.32.nsq\n",
      "refseq_genomic.33.nhr\n",
      "refseq_genomic.33.nin\n",
      "refseq_genomic.33.nnd\n",
      "refseq_genomic.33.nni\n",
      "refseq_genomic.33.nog\n",
      "refseq_genomic.33.nsd\n",
      "refseq_genomic.33.nsi\n",
      "refseq_genomic.33.nsq\n",
      "refseq_genomic.34.nhr\n",
      "refseq_genomic.34.nin\n",
      "refseq_genomic.34.nnd\n",
      "refseq_genomic.34.nni\n",
      "refseq_genomic.34.nog\n",
      "refseq_genomic.34.nsd\n",
      "refseq_genomic.34.nsi\n",
      "refseq_genomic.34.nsq\n",
      "refseq_genomic.35.nhr\n",
      "refseq_genomic.35.nin\n",
      "refseq_genomic.35.nnd\n",
      "refseq_genomic.35.nni\n",
      "refseq_genomic.35.nog\n",
      "refseq_genomic.35.nsd\n",
      "refseq_genomic.35.nsi\n",
      "refseq_genomic.35.nsq\n",
      "refseq_genomic.36.nhr\n",
      "refseq_genomic.36.nin\n",
      "refseq_genomic.36.nnd\n",
      "refseq_genomic.36.nni\n",
      "refseq_genomic.36.nog\n",
      "refseq_genomic.36.nsd\n",
      "refseq_genomic.36.nsi\n",
      "refseq_genomic.36.nsq\n",
      "refseq_genomic.37.nhr\n",
      "refseq_genomic.37.nin\n",
      "refseq_genomic.37.nnd\n",
      "refseq_genomic.37.nni\n",
      "refseq_genomic.37.nog\n",
      "refseq_genomic.37.nsd\n",
      "refseq_genomic.37.nsi\n",
      "refseq_genomic.37.nsq\n",
      "refseq_genomic.38.nhr\n",
      "refseq_genomic.38.nin\n",
      "refseq_genomic.38.nnd\n",
      "refseq_genomic.38.nni\n",
      "refseq_genomic.38.nog\n",
      "refseq_genomic.38.nsd\n",
      "refseq_genomic.38.nsi\n",
      "refseq_genomic.38.nsq\n",
      "refseq_genomic.39.nhr\n",
      "refseq_genomic.39.nin\n",
      "refseq_genomic.39.nnd\n",
      "refseq_genomic.39.nni\n",
      "refseq_genomic.39.nog\n",
      "refseq_genomic.39.nsd\n",
      "refseq_genomic.39.nsi\n",
      "refseq_genomic.39.nsq\n",
      "refseq_genomic.40.nhr\n",
      "refseq_genomic.40.nin\n",
      "refseq_genomic.40.nnd\n",
      "refseq_genomic.40.nni\n",
      "refseq_genomic.40.nog\n",
      "refseq_genomic.40.nsd\n",
      "refseq_genomic.40.nsi\n",
      "refseq_genomic.40.nsq\n",
      "refseq_genomic.41.nhr\n",
      "refseq_genomic.41.nin\n",
      "refseq_genomic.41.nnd\n",
      "refseq_genomic.41.nni\n",
      "refseq_genomic.41.nog\n",
      "refseq_genomic.41.nsd\n",
      "refseq_genomic.41.nsi\n",
      "refseq_genomic.41.nsq\n",
      "refseq_genomic.42.nhr\n",
      "refseq_genomic.42.nin\n",
      "refseq_genomic.42.nnd\n",
      "refseq_genomic.42.nni\n",
      "refseq_genomic.42.nog\n",
      "refseq_genomic.42.nsd\n",
      "refseq_genomic.42.nsi\n",
      "refseq_genomic.42.nsq\n",
      "refseq_genomic.43.nhr\n",
      "refseq_genomic.43.nin\n",
      "refseq_genomic.43.nnd\n",
      "refseq_genomic.43.nni\n",
      "refseq_genomic.43.nog\n",
      "refseq_genomic.43.nsd\n",
      "refseq_genomic.43.nsi\n",
      "refseq_genomic.43.nsq\n",
      "refseq_genomic.44.nhr\n",
      "refseq_genomic.44.nin\n",
      "refseq_genomic.44.nnd\n",
      "refseq_genomic.44.nni\n",
      "refseq_genomic.44.nog\n",
      "refseq_genomic.44.nsd\n",
      "refseq_genomic.44.nsi\n",
      "refseq_genomic.44.nsq\n",
      "refseq_genomic.45.nhr\n",
      "refseq_genomic.45.nin\n",
      "refseq_genomic.45.nnd\n",
      "refseq_genomic.45.nni\n",
      "refseq_genomic.45.nog\n",
      "refseq_genomic.45.nsd\n",
      "refseq_genomic.45.nsi\n",
      "refseq_genomic.45.nsq\n",
      "refseq_genomic.46.nhr\n",
      "refseq_genomic.46.nin\n",
      "refseq_genomic.46.nnd\n",
      "refseq_genomic.46.nni\n",
      "refseq_genomic.46.nog\n",
      "refseq_genomic.46.nsd\n",
      "refseq_genomic.46.nsi\n",
      "refseq_genomic.46.nsq\n",
      "refseq_genomic.47.nhr\n",
      "refseq_genomic.47.nin\n",
      "refseq_genomic.47.nnd\n",
      "refseq_genomic.47.nni\n",
      "refseq_genomic.47.nog\n",
      "refseq_genomic.47.nsd\n",
      "refseq_genomic.47.nsi\n",
      "refseq_genomic.47.nsq\n",
      "refseq_genomic.48.nhr\n",
      "refseq_genomic.48.nin\n",
      "refseq_genomic.48.nnd\n",
      "refseq_genomic.48.nni\n",
      "refseq_genomic.48.nog\n",
      "refseq_genomic.48.nsd\n",
      "refseq_genomic.48.nsi\n",
      "refseq_genomic.48.nsq\n",
      "refseq_genomic.49.nhr\n",
      "refseq_genomic.49.nin\n",
      "refseq_genomic.49.nnd\n",
      "refseq_genomic.49.nni\n",
      "refseq_genomic.49.nog\n",
      "refseq_genomic.49.nsd\n",
      "refseq_genomic.49.nsi\n",
      "refseq_genomic.49.nsq\n",
      "refseq_genomic.50.nhr\n",
      "refseq_genomic.50.nin\n",
      "refseq_genomic.50.nnd\n",
      "refseq_genomic.50.nni\n",
      "refseq_genomic.50.nog\n",
      "refseq_genomic.50.nsd\n",
      "refseq_genomic.50.nsi\n",
      "refseq_genomic.50.nsq\n",
      "refseq_genomic.51.nhr\n",
      "refseq_genomic.51.nin\n",
      "refseq_genomic.51.nnd\n",
      "refseq_genomic.51.nni\n",
      "refseq_genomic.51.nog\n",
      "refseq_genomic.51.nsd\n",
      "refseq_genomic.51.nsi\n",
      "refseq_genomic.51.nsq\n",
      "refseq_genomic.52.nhr\n",
      "refseq_genomic.52.nin\n",
      "refseq_genomic.52.nnd\n",
      "refseq_genomic.52.nni\n",
      "refseq_genomic.52.nog\n",
      "refseq_genomic.52.nsd\n",
      "refseq_genomic.52.nsi\n",
      "refseq_genomic.52.nsq\n",
      "refseq_genomic.53.nhr\n",
      "refseq_genomic.53.nin\n",
      "refseq_genomic.53.nnd\n",
      "refseq_genomic.53.nni\n",
      "refseq_genomic.53.nog\n",
      "refseq_genomic.53.nsd\n",
      "refseq_genomic.53.nsi\n",
      "refseq_genomic.53.nsq\n",
      "refseq_genomic.54.nhr\n",
      "refseq_genomic.54.nin\n",
      "refseq_genomic.54.nnd\n",
      "refseq_genomic.54.nni\n",
      "refseq_genomic.54.nog\n",
      "refseq_genomic.54.nsd\n",
      "refseq_genomic.54.nsi\n",
      "refseq_genomic.54.nsq\n",
      "refseq_genomic.55.nhr\n",
      "refseq_genomic.55.nin\n",
      "refseq_genomic.55.nnd\n",
      "refseq_genomic.55.nni\n",
      "refseq_genomic.55.nog\n",
      "refseq_genomic.55.nsd\n",
      "refseq_genomic.55.nsi\n",
      "refseq_genomic.55.nsq\n",
      "refseq_genomic.56.nhr\n",
      "refseq_genomic.56.nin\n",
      "refseq_genomic.56.nnd\n",
      "refseq_genomic.56.nni\n",
      "refseq_genomic.56.nog\n",
      "refseq_genomic.56.nsd\n",
      "refseq_genomic.56.nsi\n",
      "refseq_genomic.56.nsq\n",
      "refseq_genomic.57.nhr\n",
      "refseq_genomic.57.nin\n",
      "refseq_genomic.57.nnd\n",
      "refseq_genomic.57.nni\n",
      "refseq_genomic.57.nog\n",
      "refseq_genomic.57.nsd\n",
      "refseq_genomic.57.nsi\n",
      "refseq_genomic.57.nsq\n",
      "refseq_genomic.58.nhr\n",
      "refseq_genomic.58.nin\n",
      "refseq_genomic.58.nnd\n",
      "refseq_genomic.58.nni\n",
      "refseq_genomic.58.nog\n",
      "refseq_genomic.58.nsd\n",
      "refseq_genomic.58.nsi\n",
      "refseq_genomic.58.nsq\n",
      "refseq_genomic.59.nhr\n",
      "refseq_genomic.59.nin\n",
      "refseq_genomic.59.nnd\n",
      "refseq_genomic.59.nni\n",
      "refseq_genomic.59.nog\n",
      "refseq_genomic.59.nsd\n",
      "refseq_genomic.59.nsi\n",
      "refseq_genomic.59.nsq\n",
      "refseq_genomic.60.nhr\n",
      "refseq_genomic.60.nin\n",
      "refseq_genomic.60.nnd\n",
      "refseq_genomic.60.nni\n",
      "refseq_genomic.60.nog\n",
      "refseq_genomic.60.nsd\n",
      "refseq_genomic.60.nsi\n",
      "refseq_genomic.60.nsq\n",
      "refseq_genomic.61.nhr\n",
      "refseq_genomic.61.nin\n",
      "refseq_genomic.61.nnd\n",
      "refseq_genomic.61.nni\n",
      "refseq_genomic.61.nog\n",
      "refseq_genomic.61.nsd\n",
      "refseq_genomic.61.nsi\n",
      "refseq_genomic.61.nsq\n",
      "refseq_genomic.62.nhr\n",
      "refseq_genomic.62.nin\n",
      "refseq_genomic.62.nnd\n",
      "refseq_genomic.62.nni\n",
      "refseq_genomic.62.nog\n",
      "refseq_genomic.62.nsd\n",
      "refseq_genomic.62.nsi\n",
      "refseq_genomic.62.nsq\n",
      "refseq_genomic.63.nhr\n",
      "refseq_genomic.63.nin\n",
      "refseq_genomic.63.nnd\n",
      "refseq_genomic.63.nni\n",
      "refseq_genomic.63.nog\n",
      "refseq_genomic.63.nsd\n",
      "refseq_genomic.63.nsi\n",
      "refseq_genomic.63.nsq\n",
      "refseq_genomic.64.nhr\n",
      "refseq_genomic.64.nin\n",
      "refseq_genomic.64.nnd\n",
      "refseq_genomic.64.nni\n",
      "refseq_genomic.64.nog\n",
      "refseq_genomic.64.nsd\n",
      "refseq_genomic.64.nsi\n",
      "refseq_genomic.64.nsq\n",
      "refseq_genomic.65.nhr\n",
      "refseq_genomic.65.nin\n",
      "refseq_genomic.65.nnd\n",
      "refseq_genomic.65.nni\n",
      "refseq_genomic.65.nog\n",
      "refseq_genomic.65.nsd\n",
      "refseq_genomic.65.nsi\n",
      "refseq_genomic.65.nsq\n",
      "refseq_genomic.66.nhr\n",
      "refseq_genomic.66.nin\n",
      "refseq_genomic.66.nnd\n",
      "refseq_genomic.66.nni\n",
      "refseq_genomic.66.nog\n",
      "refseq_genomic.66.nsd\n",
      "refseq_genomic.66.nsi\n",
      "refseq_genomic.66.nsq\n",
      "refseq_genomic.67.nhr\n",
      "refseq_genomic.67.nin\n",
      "refseq_genomic.67.nnd\n",
      "refseq_genomic.67.nni\n",
      "refseq_genomic.67.nog\n",
      "refseq_genomic.67.nsd\n",
      "refseq_genomic.67.nsi\n",
      "refseq_genomic.67.nsq\n",
      "refseq_genomic.68.nhr\n",
      "refseq_genomic.68.nin\n",
      "refseq_genomic.68.nnd\n",
      "refseq_genomic.68.nni\n",
      "refseq_genomic.68.nog\n",
      "refseq_genomic.68.nsd\n",
      "refseq_genomic.68.nsi\n",
      "refseq_genomic.68.nsq\n",
      "refseq_genomic.69.nhr\n",
      "refseq_genomic.69.nin\n",
      "refseq_genomic.69.nnd\n",
      "refseq_genomic.69.nni\n",
      "refseq_genomic.69.nog\n",
      "refseq_genomic.69.nsd\n",
      "refseq_genomic.69.nsi\n",
      "refseq_genomic.69.nsq\n",
      "refseq_genomic.70.nhr\n",
      "refseq_genomic.70.nin\n",
      "refseq_genomic.70.nnd\n",
      "refseq_genomic.70.nni\n",
      "refseq_genomic.70.nog\n",
      "refseq_genomic.70.nsd\n",
      "refseq_genomic.70.nsi\n",
      "refseq_genomic.70.nsq\n",
      "refseq_genomic.nal\n",
      "refseq_protein.00.phr\n",
      "refseq_protein.00.pin\n",
      "refseq_protein.00.pnd\n",
      "refseq_protein.00.pni\n",
      "refseq_protein.00.pog\n",
      "refseq_protein.00.psd\n",
      "refseq_protein.00.psi\n",
      "refseq_protein.00.psq\n",
      "refseq_protein.01.phr\n",
      "refseq_protein.01.pin\n",
      "refseq_protein.01.pnd\n",
      "refseq_protein.01.pni\n",
      "refseq_protein.01.pog\n",
      "refseq_protein.01.psd\n",
      "refseq_protein.01.psi\n",
      "refseq_protein.01.psq\n",
      "refseq_protein.02.phr\n",
      "refseq_protein.02.pin\n",
      "refseq_protein.02.pnd\n",
      "refseq_protein.02.pni\n",
      "refseq_protein.02.pog\n",
      "refseq_protein.02.psd\n",
      "refseq_protein.02.psi\n",
      "refseq_protein.02.psq\n",
      "refseq_protein.03.phr\n",
      "refseq_protein.03.pin\n",
      "refseq_protein.03.pnd\n",
      "refseq_protein.03.pni\n",
      "refseq_protein.03.pog\n",
      "refseq_protein.03.psd\n",
      "refseq_protein.03.psi\n",
      "refseq_protein.03.psq\n",
      "refseq_protein.04.phr\n",
      "refseq_protein.04.pin\n",
      "refseq_protein.04.pnd\n",
      "refseq_protein.04.pni\n",
      "refseq_protein.04.pog\n",
      "refseq_protein.04.psd\n",
      "refseq_protein.04.psi\n",
      "refseq_protein.04.psq\n",
      "refseq_protein.05.phr\n",
      "refseq_protein.05.pin\n",
      "refseq_protein.05.pnd\n",
      "refseq_protein.05.pni\n",
      "refseq_protein.05.pog\n",
      "refseq_protein.05.psd\n",
      "refseq_protein.05.psi\n",
      "refseq_protein.05.psq\n",
      "refseq_protein.06.phr\n",
      "refseq_protein.06.pin\n",
      "refseq_protein.06.pnd\n",
      "refseq_protein.06.pni\n",
      "refseq_protein.06.pog\n",
      "refseq_protein.06.psd\n",
      "refseq_protein.06.psi\n",
      "refseq_protein.06.psq\n",
      "refseq_protein.07.phr\n",
      "refseq_protein.07.pin\n",
      "refseq_protein.07.pnd\n",
      "refseq_protein.07.pni\n",
      "refseq_protein.07.pog\n",
      "refseq_protein.07.psd\n",
      "refseq_protein.07.psi\n",
      "refseq_protein.07.psq\n",
      "refseq_protein.08.phr\n",
      "refseq_protein.08.pin\n",
      "refseq_protein.08.pnd\n",
      "refseq_protein.08.pni\n",
      "refseq_protein.08.pog\n",
      "refseq_protein.08.psd\n",
      "refseq_protein.08.psi\n",
      "refseq_protein.08.psq\n",
      "refseq_protein.pal\n",
      "refseq_rna.00.nhr\n",
      "refseq_rna.00.nin\n",
      "refseq_rna.00.nnd\n",
      "refseq_rna.00.nni\n",
      "refseq_rna.00.nog\n",
      "refseq_rna.00.nsd\n",
      "refseq_rna.00.nsi\n",
      "refseq_rna.00.nsq\n",
      "refseq_rna.01.nhr\n",
      "refseq_rna.01.nin\n",
      "refseq_rna.01.nnd\n",
      "refseq_rna.01.nni\n",
      "refseq_rna.01.nog\n",
      "refseq_rna.01.nsd\n",
      "refseq_rna.01.nsi\n",
      "refseq_rna.01.nsq\n",
      "refseq_rna.02.nhr\n",
      "refseq_rna.02.nin\n",
      "refseq_rna.02.nnd\n",
      "refseq_rna.02.nni\n",
      "refseq_rna.02.nog\n",
      "refseq_rna.02.nsd\n",
      "refseq_rna.02.nsi\n",
      "refseq_rna.02.nsq\n",
      "refseq_rna.nal\n",
      "refseqgene.nhr\n",
      "refseqgene.nin\n",
      "refseqgene.nnd\n",
      "refseqgene.nni\n",
      "refseqgene.nog\n",
      "refseqgene.nsd\n",
      "refseqgene.nsi\n",
      "refseqgene.nsq\n",
      "swissprot.00.msk\n",
      "swissprot.00.pal\n",
      "swissprot.01.msk\n",
      "swissprot.01.pal\n",
      "swissprot.02.msk\n",
      "swissprot.02.pal\n",
      "swissprot.03.msk\n",
      "swissprot.03.pal\n",
      "swissprot.04.msk\n",
      "swissprot.04.pal\n",
      "swissprot.05.msk\n",
      "swissprot.05.pal\n",
      "swissprot.06.msk\n",
      "swissprot.06.pal\n",
      "swissprot.07.msk\n",
      "swissprot.07.pal\n",
      "swissprot.08.msk\n",
      "swissprot.08.pal\n",
      "swissprot.09.msk\n",
      "swissprot.09.pal\n",
      "swissprot.10.msk\n",
      "swissprot.10.pal\n",
      "swissprot.11.msk\n",
      "swissprot.11.pal\n",
      "swissprot.12.msk\n",
      "swissprot.12.pal\n",
      "swissprot.13.msk\n",
      "swissprot.13.pal\n",
      "swissprot.pal\n",
      "taxdb.btd\n",
      "taxdb.bti\n",
      "taxdb.tar.gz\n",
      "uniprot-mouse-reference.fasta\n",
      "uniprot-mouse-reference.phr\n",
      "uniprot-mouse-reference.pin\n",
      "uniprot-mouse-reference.psq\n",
      "uniprot_sprot.fasta\n",
      "uniprot_sprot.phr\n",
      "uniprot_sprot.pin\n",
      "uniprot_sprot.psq\n",
      "uniprot_sprot_091616.fasta\n",
      "uniprot_sprot_091616.phr\n",
      "uniprot_sprot_091616.pin\n",
      "uniprot_sprot_091616.psq\n",
      "vector.nhr\n",
      "vector.nin\n",
      "vector.nnd\n",
      "vector.nni\n",
      "vector.nsd\n",
      "vector.nsi\n",
      "vector.nsq\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /Volumes/Data/blast_db/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/srlab/gitrepos/LabDocs/jupyter_nbs/grace\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process is interrupted.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "blastp \\\n",
    "-query /Volumes/web/scaphapoda/Grace/Transcriptomes/c_gigas/query.fa \\\n",
    "-db /Volumes/Data/blast_db/uniprot_sprot \\\n",
    "-max_target_seqs 1 \\\n",
    "-max_hsps 1 \\\n",
    "-outfmt 6 \\\n",
    "-num_threads 16 \\\n",
    "-out /Volumes/toaster/grace/blast_gigas_sprot.tab "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">CHOYP_043R.1.5|m.16874\n",
      "TPSGPTPSGPTPSVTPTPSGPTPSVTPTPSGSTPSGPTPSVTPTPSGPTPSGPTPSVTPT\n",
      "PSGPTPSVTPTPSVPTPSGPTPSVTPTPSGPTPSVTPTPSGPTPSGPTPSVTPTPSGPTP\n",
      "SGPTPSVTPTPSVTPTPSGPTPSVTSTPSAPTPSGPTPSGPTPSVTPTPSGPTPSGPTPS\n",
      "VTPTPSGPTPSVTPTPSG\n",
      ">CHOYP_043R.5.5|m.64252\n",
      "SRPTPSVTPTPSGPTPSVTPTPSVSTPSGPTPSVTPTPSGPSPSVTPTPSGPSPSGPTPS\n",
      "ATPTPSGPTPSGTTPSGSTPSATITTISTPSTTVCSYVDIGPEQAIDVSLRSPSEDPDAP\n",
      "IENILQTNSVYKPKKEPTYDENVVVKIISQDTPTILRVSFTVNRADTVGLEYLTDYKQKI\n",
      "ITQNNETVEFVFAAGIITDNFTINIRSDSAEQPEISNLKIRACYKPVIGQPSTTTPNPSI\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head /Volumes/web/Grace/Transcriptomes/c_gigas/query.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHOYP_043R.5.5|m.64252\tsp|Q06852|SLAP1_CLOTH\t57.45\t235\t61\t19\t611\t816\t1467\t1691\t2e-15\t85.1\n",
      "CHOYP_14332.1.2|m.5643\tsp|Q2F637|1433Z_BOMMO\t66.03\t262\t74\t2\t19\t280\t1\t247\t1e-117\t  344\n",
      "CHOYP_14332.1.2|m.5644\tsp|P62325|BTG1_MOUSE\t47.20\t161\t80\t2\t1\t156\t11\t171\t4e-46\t  155\n",
      "CHOYP_14332.2.2|m.61737\tsp|P68250|1433B_BOVIN\t66.80\t250\t79\t1\t1\t250\t1\t246\t4e-115\t  337\n",
      "CHOYP_14332.2.2|m.61738\tsp|Q2PQA9|KINH_RAT\t32.76\t116\t64\t3\t33\t136\t646\t759\t0.20\t36.2\n",
      "CHOYP_1433E.1.2|m.3639\tsp|Q9CWP8|DPOD4_MOUSE\t38.24\t102\t61\t2\t30\t130\t7\t107\t7e-18\t78.6\n",
      "CHOYP_1433E.1.2|m.3638\tsp|P62262|1433E_SHEEP\t80.31\t254\t49\t1\t1\t254\t1\t253\t5e-146\t  416\n",
      "CHOYP_1433E.2.2|m.63376\tsp|P92177|1433E_DROME\t77.69\t260\t53\t1\t1\t255\t1\t260\t1e-147\t  420\n",
      "CHOYP_1433E.2.2|m.63378\tsp|Q6MG82|PRRT1_RAT\t44.44\t99\t53\t1\t41\t137\t196\t294\t3e-18\t82.8\n",
      "CHOYP_1433E.2.2|m.63377\tsp|Q18CI9|ECFA2_CLOD6\t48.57\t35\t18\t0\t201\t235\t19\t53\t0.008\t40.4\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head /Volumes/toaster/grace/blast_gigas_sprot.tab "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cp /Volumes/web/scaphapoda/Grace/Transcriptomes/c_gigas/blast_gigas_sprot.tab \\\n",
    "/Volumes/toaster/grace/blast_gigas_sprot.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "tr '|' \"\\t\" </Volumes/toaster/grace/blast_gigas_sprot.tab> \\\n",
    "/Volumes/toaster/grace/blast_gigas_sprot_3.tab \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     770\r\n"
     ]
    }
   ],
   "source": [
    "!cat /Volumes/toaster/grace/blast_gigas_sprot_3.tab | wc -l\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHOYP_043R.5.5\tm.64252\tsp\tQ06852\tSLAP1_CLOTH\t57.45\t235\t61\t19\t611\t816\t1467\t1691\t2e-15\t85.1\r\n",
      "CHOYP_14332.1.2\tm.5643\tsp\tQ2F637\t1433Z_BOMMO\t66.03\t262\t74\t2\t19\t280\t1\t247\t1e-117\t  344\r\n",
      "CHOYP_14332.1.2\tm.5644\tsp\tP62325\tBTG1_MOUSE\t47.20\t161\t80\t2\t1\t156\t11\t171\t4e-46\t  155\r\n",
      "CHOYP_14332.2.2\tm.61737\tsp\tP68250\t1433B_BOVIN\t66.80\t250\t79\t1\t1\t250\t1\t246\t4e-115\t  337\r\n",
      "CHOYP_14332.2.2\tm.61738\tsp\tQ2PQA9\tKINH_RAT\t32.76\t116\t64\t3\t33\t136\t646\t759\t0.20\t36.2\r\n",
      "CHOYP_1433E.1.2\tm.3639\tsp\tQ9CWP8\tDPOD4_MOUSE\t38.24\t102\t61\t2\t30\t130\t7\t107\t7e-18\t78.6\r\n",
      "CHOYP_1433E.1.2\tm.3638\tsp\tP62262\t1433E_SHEEP\t80.31\t254\t49\t1\t1\t254\t1\t253\t5e-146\t  416\r\n",
      "CHOYP_1433E.2.2\tm.63376\tsp\tP92177\t1433E_DROME\t77.69\t260\t53\t1\t1\t255\t1\t260\t1e-147\t  420\r\n",
      "CHOYP_1433E.2.2\tm.63378\tsp\tQ6MG82\tPRRT1_RAT\t44.44\t99\t53\t1\t41\t137\t196\t294\t3e-18\t82.8\r\n",
      "CHOYP_1433E.2.2\tm.63377\tsp\tQ18CI9\tECFA2_CLOD6\t48.57\t35\t18\t0\t201\t235\t19\t53\t0.008\t40.4\r\n"
     ]
    }
   ],
   "source": [
    "!head /Volumes/toaster/grace/blast_gigas_sprot_3.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "SELECT column1, column4 FROM graceac9.\"blast_gigas_sprot_2.tab\"blast\n",
    "LEFT JOIN sr320.\"uniprot-reviewed-annotations_041017\"uni\n",
    "ON blast.column4 = uni.\"Entry\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "SELECT * FROM graceac9.\"blast_gigas_sprot_2.tab\"blast\n",
    "LEFT JOIN sr320.\"SPID_and_GO_number\"ggo\n",
    "ON blast.column4 = ggo.column1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "SELECT * FROM graceac9.\"blast_gigas_sprot_3.tab\"blast\n",
    "LEFT JOIN sr320.\"uniprot-reviewed-annotations_041017\"uni\n",
    "ON blast.column4 = uni.\"Entry\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "col1\tcolumn2\tcolumn3\tcolumn4\tGene ontology IDs\r\n",
      "CHOYP_ACAC.2.2\tm.58115\tsp\tB5ZV92\tGO:0000105; GO:0003949; GO:0005737\r\n",
      "CHOYP_ACSS3.1.1\tm.64145\tsp\tQ7TMK9\tGO:0000166; GO:0001649; GO:0003723; GO:0005654; GO:0005737; GO:0005783; GO:0006397; GO:0008143; GO:0008380; GO:0016020; GO:0017148; GO:0030529; GO:0070934; GO:0070937; GO:0071013; GO:0071204; GO:0071346; GO:0097452\r\n",
      "CHOYP_AGAP_AGAP010186.1.1\tm.62503\tsp\tQ8NDA2\tGO:0005509; GO:0005604; GO:0005938; GO:0006939; GO:0030054; GO:0030335; GO:0032154; GO:0050896\r\n",
      "CHOYP_AGAP_AGAP010742.1.1\tm.51062\tsp\tA2AJ76\tGO:0005509; GO:0005576; GO:0005604; GO:0005938; GO:0006939; GO:0030054; GO:0030335; GO:0032154; GO:0050896\r\n",
      "CHOYP_ACA1_370240.1.1\tm.10015\tsp\tQ5HZL9\tGO:0008152; GO:0016787\r\n",
      "CHOYP_A1M.1.2\tm.39785\tsp\tP95949\tGO:0003676; GO:0004004; GO:0005524; GO:0010501\r\n",
      "CHOYP_AAEL_AAEL003853.2.2\tm.61199\tsp\tQ96RW7\tGO:0005509; GO:0005604; GO:0005938; GO:0007049; GO:0007601; GO:0030054; GO:0032154; GO:0050896; GO:0051301; GO:0070062\r\n",
      "CHOYP_ADAM20.1.2\tm.16195\tsp\tQ757T0\tGO:0004842; GO:0031499\r\n",
      "CHOYP_AAEL_AAEL007787.1.1\tm.13188\tsp\tP13195\tGO:0001666; GO:0003870; GO:0005739; GO:0005759; GO:0006782; GO:0009635; GO:0010033; GO:0010045; GO:0014070; GO:0030170; GO:0031667; GO:0032025; GO:0032869; GO:0034698; GO:0042493; GO:0045471; GO:0051591; GO:0070541; GO:0071407\r\n"
     ]
    }
   ],
   "source": [
    "!head /Volumes/toaster/grace/blast-join-GO.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "# This script was written to address a specific problem that Rhonda was having\n",
    "# where multiple GO terms are present in a single field, but she needs them\n",
    "# on separate lines.\n",
    "# See this GitHub issue for explanation of the problem:\n",
    "# https://github.com/sr320/LabDocs/issues/654\n",
    "\n",
    "# This script is written to operate on the file located here:\n",
    "# https://github.com/RobertsLab/project-pacific.oyster-larvae/blob/master/DIA_2015/AnnotatedproteinsGO.tabular\n",
    "\n",
    "# Before executing this script, the following need to be changed (look for them below):\n",
    "\n",
    "# /path/to/input/file\n",
    "# /path/to/new/filename\n",
    "# /path/to/new/filename\n",
    "\n",
    "# sed command substitutes the \"; \" sequence to a tab and writes the new format to a new file.\n",
    "# This character sequence is how the GO terms are delimited in their field.\n",
    "tr '; ' \"\\t\" </Volumes/toaster/grace/blast-join-GO.txt> /Volumes/toaster/grace/blast-join-GO-unfold.txt\n",
    "\n",
    "# Saves location of the newly formatted file to variable called \"file\".\n",
    "file=\"/Volumes/toaster/grace/blast-join-GO-unfold.txt\"\n",
    "\n",
    "# Identify first field containing a GO term.\n",
    "# Search file with grep for \"GO:\" and pipe to awk.\n",
    "# Awk sets tab as field delimiter (-F'\\t'), runs a for loop that looks for \"GO:\" (~/GO:/), and then prints the field number).\n",
    "# Awk results are piped to sort, which sorts unique by number (-ug).\n",
    "# Sort results are piped to head to retrieve the lowest value (i.e. the top of the list; \"-n1\").\n",
    "begin_goterms=$(grep \"GO:\" \"$file\" | awk -F'\\t' '{for (i=1;i<=NF;i++) if($i ~/GO:/) print i}' | sort -ug | head -n1)\n",
    "\n",
    "# While loop to process each line of the input file.\n",
    "while read -r line\n",
    "\tdo\n",
    "\t\n",
    "\t# Send contents of the current line to awk.\n",
    "\t# Set the field separator as a tab (-F'\\t') and print the number of fields in that line.\n",
    "\t# Save the results of the echo/awk pipe (i.e. number of fields) to the variable \"max_field\".\n",
    "\tmax_field=$(echo \"$line\" | awk -F'\\t' '{print NF}')\n",
    "\n",
    "\t# Send contents of current line to cut.\n",
    "\t# Cut fields (i.e. retain those fields) 1-12.\n",
    "\t# Save the results of the echo/cut pipe (i.e. fields 1-12) to the variable \"fixed_fields\"\n",
    "\tfixed_fields=$(echo \"$line\" | cut -f1-12)\n",
    "\n",
    "\t# Since not all the lines contain the same number of fields (e.g. may not have GO terms),\n",
    "\t# evaluate the number of fields in each line to determine how to handle current line.\n",
    "\n",
    "\t# If the value in max_field is less than the field number where the GO terms begin,\n",
    "\t# then just print the current line (%s) followed by a newline (\\n).\n",
    "\tif (( \"$max_field\" < \"$begin_goterms\" ))\n",
    "\t\tthen printf \"%s\\n\" \"$line\"\n",
    "\t\t\telse\n",
    "\n",
    "\t\t\t# Send contents of current line (which contains GO terms) to cut.\n",
    "\t\t\t# Cut fields (i.e. retain those fields) 13 to whatever the last field is in the curent line.\n",
    "\t\t\t# Save the results of the echo/cut pipe (i.e. all the GO terms fields) to the variable \"goterms\".\n",
    "\t\t\tgoterms=$(echo \"$line\" | cut -f\"$begin_goterms\"-\"$max_field\")\n",
    "\t\t\t\n",
    "\t\t\t# Assign values in the variable \"goterms\" to a new indexed array (called \"array\"), \n",
    "\t\t\t# with tab delimiter (IFS=$'\\t')\n",
    "\t\t\tIFS=$'\\t' read -r -a array <<<\"$goterms\"\n",
    "\t\t\t\n",
    "\t\t\t# Iterate through each element of the array.\n",
    "\t\t\t# Print the first 12 fields (i.e. the fields stored in \"fixed_fields\") followed by a tab (%s\\t).\n",
    "\t\t\t# Print the current element in the array (i.e. the current GO term) followed by a new line (%s\\n).\n",
    "\t\t\tfor element in \"${!array[@]}\"\t\n",
    "\t\t\t\tdo printf \"%s\\t%s\\n\" \"$fixed_fields\" \"${array[$element]}\"\n",
    "\t\t\tdone\n",
    "\tfi\n",
    "\n",
    "# Send the input file into the while loop and send the output to a file named \"rhonda_fixed.txt\".\n",
    "done < \"$file\" > /Volumes/toaster/grace/blast-join-GO-unfold-final-final.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "col1\tcolumn2\tcolumn3\tcolumn4\tGene\tontology\tIDs\tGene\r\n",
      "col1\tcolumn2\tcolumn3\tcolumn4\tGene\tontology\tIDs\tontology\r\n",
      "col1\tcolumn2\tcolumn3\tcolumn4\tGene\tontology\tIDs\tIDs\r\n",
      "CHOYP_ACAC.2.2\tm.58115\tsp\tB5ZV92\tGO:0000105\t\tGO:0003949\t\tGO:0005737\tGO:0000105\r\n",
      "CHOYP_ACAC.2.2\tm.58115\tsp\tB5ZV92\tGO:0000105\t\tGO:0003949\t\tGO:0005737\tGO:0003949\r\n",
      "CHOYP_ACAC.2.2\tm.58115\tsp\tB5ZV92\tGO:0000105\t\tGO:0003949\t\tGO:0005737\tGO:0005737\r\n",
      "CHOYP_ACSS3.1.1\tm.64145\tsp\tQ7TMK9\tGO:0000166\t\tGO:0001649\t\tGO:0003723\t\tGO:0005654\t\tGO:0000166\r\n",
      "CHOYP_ACSS3.1.1\tm.64145\tsp\tQ7TMK9\tGO:0000166\t\tGO:0001649\t\tGO:0003723\t\tGO:0005654\t\tGO:0001649\r\n",
      "CHOYP_ACSS3.1.1\tm.64145\tsp\tQ7TMK9\tGO:0000166\t\tGO:0001649\t\tGO:0003723\t\tGO:0005654\t\tGO:0003723\r\n",
      "CHOYP_ACSS3.1.1\tm.64145\tsp\tQ7TMK9\tGO:0000166\t\tGO:0001649\t\tGO:0003723\t\tGO:0005654\t\tGO:0005654\r\n"
     ]
    }
   ],
   "source": [
    "!head /Volumes/toaster/grace/blast-join-GO-unfold-final-final.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "tr '; ' \"\\t\" </Volumes/toaster/grace/blast-join-GO.txt> /Volumes/toaster/grace/blast-join-GO-3.txt\n",
    "\n",
    "file=\"/Volumes/toaster/grace/blast-join-GO-3.txt\"\n",
    "while read -r line\n",
    "\tdo\n",
    "\tmax_field=$(echo \"$line\" | awk -F'\\t' '{print NF}')\n",
    "\tset_fields=$(echo \"$line\" | cut -f1-4)\n",
    "\tif (( \"$max_field\" < 5 ))\n",
    "\t\tthen printf \"%s\\n\" \"$line\"\n",
    "\t\telse\n",
    "\t\t\tgoterms=$(echo \"$line\" | cut -f5-$max_field)\n",
    "\t\t\tIFS=$'\\t' read -r -a array <<<\"$goterms\"\n",
    "\t\t\t\tfor element in \"${!array[@]}\"\t\n",
    "\t\t\t\t\tdo printf \"%s\\t%s\\n\" \"$set_fields\" \"${array[$element]}\"\n",
    "\t\t\t\tdone\n",
    "\tfi\n",
    "done < \"$file\" > /Volumes/toaster/grace/blast-join-GO-final-45.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "col1\tcolumn2\tcolumn3\tcolumn4\tGene\r\n",
      "col1\tcolumn2\tcolumn3\tcolumn4\tontology\r\n",
      "col1\tcolumn2\tcolumn3\tcolumn4\tIDs\r\n",
      "CHOYP_ACAC.2.2\tm.58115\tsp\tB5ZV92\tGO:0000105\r\n",
      "CHOYP_ACAC.2.2\tm.58115\tsp\tB5ZV92\tGO:0003949\r\n",
      "CHOYP_ACAC.2.2\tm.58115\tsp\tB5ZV92\tGO:0005737\r\n",
      "CHOYP_ACSS3.1.1\tm.64145\tsp\tQ7TMK9\tGO:0000166\r\n",
      "CHOYP_ACSS3.1.1\tm.64145\tsp\tQ7TMK9\tGO:0001649\r\n",
      "CHOYP_ACSS3.1.1\tm.64145\tsp\tQ7TMK9\tGO:0003723\r\n",
      "CHOYP_ACSS3.1.1\tm.64145\tsp\tQ7TMK9\tGO:0005654\r\n"
     ]
    }
   ],
   "source": [
    "!head /Volumes/toaster/grace/blast-join-GO-final-45.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
